tsk/fs/hfs.cpp

```diff
@@ -746,16 +730,17 @@ hfs_cat_traverse(HFS_INFO * hfs,
 {
     TSK_FS_INFO *fs = &(hfs->fs_info);
     uint32_t cur_node;          /* node id of the current node */
-    char *node;
 
     uint16_t nodesize;
     uint8_t is_done = 0;
 
     tsk_error_reset();
 
     nodesize = tsk_getu16(fs->endian, hfs->catalog_header.nodesize);
-    if ((node = (char *) tsk_malloc(nodesize)) == NULL)
+    std::unique_ptr<char[]> node{new char[nodesize]};
+    if (!node) {
         return 1;
+    }
```

Comment:
can this even fail? I thought that a failed `new` would throw an exception.

Reply:
It will. These should all be the nothrow version of `new`.